driver-sync/src/test/functional/com/mongodb/client/ClientSideEncryptionViewAreProhibitedTest.java

```diff
@@ -0,0 +1,106 @@
+import static org.junit.Assume.assumeTrue;
+
+public class ClientSideEncryptionViewAreProhibitedTest {
+    private MongoClient client, clientEncrypted;
+
+    @Before
+    public void setUp() {
+        assumeFalse(isNotAtLeastJava8());
+        assumeTrue(serverVersionAtLeast(4, 1));
+        assumeTrue("Encryption test with external keyVault is disabled",
+                System.getProperty("org.mongodb.test.awsAccessKeyId") != null
+                        && !System.getProperty("org.mongodb.test.awsAccessKeyId").isEmpty());
+
+        MongoClientSettings.Builder clientSettingsBuilder = getMongoClientSettingsBuilder();
+        CodecRegistry codecRegistry = fromRegistries(
+                fromCodecs(new UuidCodec(UuidRepresentation.STANDARD)),
+                MongoClientSettings.getDefaultCodecRegistry());
+        clientSettingsBuilder.codecRegistry(codecRegistry);
+        MongoClientSettings clientSettings = clientSettingsBuilder.build();
+        client = MongoClients.create(clientSettings);
+
+        MongoDatabase db = client.getDatabase("db");
+        db.getCollection("view").drop();
+
+        BsonDocument createCmd = new BsonDocument()
+                .append("create", new BsonString("view"))
+                .append("viewOn", new BsonString("coll"));
+
+        db.runCommand(createCmd, BsonDocument.class);
+
+        Map<String, Map<String, Object>> kmsProviders = new HashMap<String, Map<String, Object>>();
+        Map<String, Object> localMasterkey = new HashMap<String, Object>();
+
+        byte[] localMasterkeyBytes = Base64.getDecoder().decode("Mng0NCt4ZHVUYUJCa1kxNkVyNUR1QURhZ2h2UzR2d2RrZzh0cFBwM3R6NmdWMDFBM"
+                + "UN3YkQ5aXRRMkhGRGdQV09wOGVNYUMxT2k3NjZKelhaQmRCZGJkTXVyZG9uSjFk");
+        localMasterkey.put("key", localMasterkeyBytes);
+        kmsProviders.put("local", localMasterkey);
+
+        AutoEncryptionSettings.Builder autoEncryptionSettingsBuilder = AutoEncryptionSettings.builder()
+                .keyVaultNamespace("admin.datakeys")
+                .kmsProviders(kmsProviders);
+
+        AutoEncryptionSettings autoEncryptionSettings = autoEncryptionSettingsBuilder.build();
+
+        clientSettings = clientSettingsBuilder
+                .codecRegistry(codecRegistry)
+                .autoEncryptionSettings(autoEncryptionSettings)
+                .build();
+        clientEncrypted = MongoClients.create(clientSettings);
+    }
+
+    @Test
+    public void shouldThrowError() {
+        boolean exceptionThrown = false;
+        MongoCollection<BsonDocument> coll = clientEncrypted
+                .getDatabase("db")
+                .getCollection("view", BsonDocument.class);
+        try {
+            coll.insertOne(new BsonDocument().append("encrypted", new BsonString("test")));
+        } catch (MongoException me) {
+            exceptionThrown = true;
+            assertTrue(me.getMessage().contains("cannot auto encrypt a view"));
+        }
+        assertTrue(exceptionThrown);
```

Comment:
The more traditional way of doing this is to put a call to `fail()` inside the try block, right after the call to `insertOne`.  This will assert when called, which is what you want. Then you can remove the boolean and this assertion.